apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -874,10 +881,12 @@ def _coo_to_table(mat_coo: sp.coo_matrix, axis: int = 0, base: int = 0) -> pa.Ta
         t1 = time.time()
 
         # Chunk size on the stride axis
-        if isinstance(matrix, np.ndarray):
+        if isinstance(matrix, (np.ndarray, h5py.Dataset)):
```

Comment:
Based on some past work, I'm not sure this is a reliable way to detect the type of the X matrix as you are using an implementation detail of AnnData. There is a way to detect this using the offiial API - just index/slice the X array and you will get one of the declared return types (ndarray, scipy.sparse).

I don't think it is critical to fix this in this PR, but the above may bite us if AnnData evolves (and I believe 0.9 is around the corner).

Reply:
BTW, I have code that does just this located [here](https://github.com/chanzuckerberg/single-cell-curation/blob/4097e0c370a675917f3617b1c9ce057ddc021609/cellxgene_schema_cli/cellxgene_schema/validate.py#L393)

Reply:
> just index/slice the X array
this will cause an OOM :) 

maybe slicing by `[0]`

Reply:
Definitely a great idea and definitely for a separate PR -- thank you @bkmartinjr !

Reply:
#1064